contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -472,6 +473,30 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
         }
     }
 
+    /**
+     * @dev Remove just enough liquidity so that the swap operation can be performed. No need to
+     * add the gaugeUnstakeAndRestake modifier since the underlying _removeLiquidity function
+     * will do that.
+     * Good to consider that remaining liquidity will remain in the pool & staked to the gauge.
+     *
+     * @param _amountToSwap The amount of the token to swap
+     * @param _swapWeth Swap using WETH when true, use OETHb when false
+     */
+    function _removeLiquidityToEnsureSwap(uint256 _amountToSwap, bool _swapWeth)
```

Comment:
⚪️ I think we should get rid of this function. The only thing it's adding over calling ` _ensureWETHBalance` is toto check two conditions, and burnOethB afterwords. But this is the wrong place to burn superOETHb, and both the conditions can easily be checked in the one if statement around the one place this is called.

Reply:
agreed adding another call adds more complexity over checking for these conditions in the function that is calling it. Removing this function

Reply:
done here: https://github.com/OriginProtocol/origin-dollar/pull/2276/commits/3bd3be1a8f5ab249153a9e90456c2126b0b18788

Reply:
Looks good.